sdk/ai/azure-ai-agents/azure/ai/agents/telemetry/_ai_agents_instrumentor.py

```diff
@@ -1352,125 +1636,53 @@ def trace_list_messages(self, function, *args, **kwargs):
         server_address = self.get_server_address_from_arg(args[0])
         thread_id = kwargs.get("thread_id")
 
-        span = self.start_list_messages_span(server_address=server_address, thread_id=thread_id)
-
-        if span is None:
-            return function(*args, **kwargs)
-
-        with span:
-            try:
-                result = function(*args, **kwargs)
-                for message in result.data:
-                    self.add_thread_message_event(span, message)
-
-            except Exception as exc:
-                # Set the span status to error
-                if isinstance(span.span_instance, Span):  # pyright: ignore [reportPossiblyUnboundVariable]
-                    span.span_instance.set_status(
-                        StatusCode.ERROR,  # pyright: ignore [reportPossiblyUnboundVariable]
-                        description=str(exc),
-                    )
-                module = getattr(exc, "__module__", "")
-                module = module if module != "builtins" else ""
-                error_type = f"{module}.{type(exc).__name__}" if module else type(exc).__name__
-                self._set_attributes(span, ("error.type", error_type))
-                raise
+        span = self.start_list_messages_span(
+            server_address=server_address, thread_id=thread_id
+        )
 
-        return result
+        return _InstrumentedPageableList(
+            function(*args, **kwargs), self.add_thread_message_event, span
+        )
 
-    async def trace_list_messages_async(self, function, *args, **kwargs):
+    def trace_list_messages_async(self, function, *args, **kwargs):
+        # Note that this method is not async, but it operates on AsyncIterable.
         server_address = self.get_server_address_from_arg(args[0])
         thread_id = kwargs.get("thread_id")
 
-        span = self.start_list_messages_span(server_address=server_address, thread_id=thread_id)
-
-        if span is None:
-            return await function(*args, **kwargs)
-
-        with span:
-            try:
-                result = await function(*args, **kwargs)
-                for message in result.data:
-                    self.add_thread_message_event(span, message)
-
-            except Exception as exc:
-                # Set the span status to error
-                if isinstance(span.span_instance, Span):  # pyright: ignore [reportPossiblyUnboundVariable]
```

Comment:
How this part is done in the new implementation?

Reply:
We call the same error handling code in _SpanLoggableMixin